imblearn/under_sampling/_prototype_selection/_one_sided_selection.py

```diff
@@ -115,14 +115,8 @@ def _validate_estimator(self):
             self.estimator_ = KNeighborsClassifier(
                 n_neighbors=self.n_neighbors, n_jobs=self.n_jobs
             )
-        elif isinstance(self.n_neighbors, KNeighborsClassifier):
-            self.estimator_ = clone(self.n_neighbors)
         else:
-            raise ValueError(
-                f"`n_neighbors` has to be a int or an object"
-                f" inherited from KNeighborsClassifier."
-                f" Got {type(self.n_neighbors)} instead."
-            )
+            self.estimator_ = clone(self.n_neighbors)
```

Comment:
This change removes the `isinstance` check for `OneSidedSelection`, similar to how [`KMeansSMOTE`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/f177b05256fa3e4e556c1a8be0283b721b1bcd86/imblearn/over_sampling/_smote/cluster.py#L148-L158) does not require such a check.